kubejs/server_scripts/base/entity_events/hurt/void_safety_net.js

```diff
@@ -1,8 +1,15 @@
 EntityEvents.hurt((event) => {
     const { entity, source, level, server } = event;
-    if (!entity.isPlayer() || entity.isFake()) {
+    if (!entity.isPlayer() || entity.isFake()) return;
+
+    let currentTime = level.getDayTime();
+    if (
+        server.persistentData['lastVoidTime_' + entity.getUsername()] &&
+        server.persistentData['lastVoidTime_' + entity.getUsername()] + 40 > currentTime
```

Comment:
Looks good to me. Would just like @NielsPilgaard's input here on whether it should be stored on server or user persistentData.

My thinking is server makes more sense, since the issue you were experience seems to be due to latency, so trying to push to the player in a high latency situation wouldn't necessarily work in time either. Storing player specific data on the player itself does seem to make more sense in general, otherwise.

Reply:
actually, maybe I'm just bananas either way since the player data would still be stored server side...

Reply:
I'd prefer to have this on the player's data, if it's not annoying to implement 😄